source/common/common/cleanup.h

```diff
@@ -55,4 +55,39 @@ template <class T> class RaiiListElement {
   bool cancelled_;
 };
 
+template <class Key, class Value> class RaiiMapOfListElement {
```

Comment:
Please add a block comment about what this class does.

Reply:
Done in 7862a45